src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/SecretBuildWrapper.java

```diff
@@ -72,12 +85,44 @@
         };
     }
 
+    @Override public OutputStream decorateLogger(AbstractBuild build, OutputStream logger)
+            throws IOException, InterruptedException, RunnerAbortedException {
+
+        Map<String,String> overrides = new HashMap<String, String>();
+        for (MultiBinding<?> binding : bindings) {
+            MultiBinding.MultiEnvironment environment = binding.bind(build, null, null, null);
+            build.addAction(new SharedMultiEnvAction(environment));
+            for (String envKey : environment.getValues().keySet()) {
+                if (!environment.getValues().get(envKey).isEmpty()) {
+                    overrides.put(envKey, environment.getValues().get(envKey));
+                }
+            }
+        }
+        if (!overrides.isEmpty()) {
+            return new BindingStep.Filter(overrides.values(), build.getCharset().name()).decorateLogger(build, logger);
+        } else {
+            return logger;
+        }
+    }
+
     @Override public void makeSensitiveBuildVariables(AbstractBuild build, Set<String> sensitiveVariables) {
         for (MultiBinding binding : bindings) {
             sensitiveVariables.addAll(binding.variables());
         }
     }
 
+    private static class SharedMultiEnvAction extends InvisibleAction {
+        private final transient MultiBinding.MultiEnvironment me;
```

Comment:
Adding an `Action` with only a `transient` field is pointless. Better to just keep a `WeakHashMap`.